Modules/posixmodule.c

```diff
@@ -1284,27 +1294,37 @@ path_converter(PyObject *o, void *p)
     }
 
     if (is_unicode) {
+#ifndef MS_WINDOWS
+        if (path->make_wide) {
+#endif
+            wide = PyUnicode_AsWideCharString(o, &length);
+            if (!wide) {
+                goto error_exit;
+            }
 #ifdef MS_WINDOWS
-        wide = PyUnicode_AsWideCharString(o, &length);
-        if (!wide) {
-            goto error_exit;
-        }
-        if (length > 32767) {
-            FORMAT_EXCEPTION(PyExc_ValueError, "%s too long for Windows");
-            goto error_exit;
-        }
-        if (wcslen(wide) != length) {
-            FORMAT_EXCEPTION(PyExc_ValueError, "embedded null character in %s");
-            goto error_exit;
-        }
+            if (length > 32767) {
+                FORMAT_EXCEPTION(PyExc_ValueError, "%s too long for Windows");
+                goto error_exit;
+            }
```

Comment:
An option is needed to gate this check since `splitroot()` and `normpath()` shouldn't be limited to the path length limit of the NT kernel. Maybe add a default-enabled `raise_value_error` option to allow raising `ValueError`. This would also apply to Unicode errors for embedded null characters, so it would replace the `null_embeddable` option.

Reply:
Though why this check has to be here at all isn't obvious to me. We don't limit the path length on POSIX platforms.